cyder/base/fields.py

```diff
@@ -51,9 +51,9 @@ def to_python(self, value):
         value = super(MacAddrField, self).to_python(value)
 
         if value:
-            value = value.lower().replace(':', '').replace('-', '')
+            raw = value.lower().replace(':', '').replace('-', '')
             value = reduce(lambda x,y: x + ':' + y,
-                           (value[i:i+2] for i in xrange(0, 12, 2)))
+                           (raw[i:i+2] for i in xrange(0, 12, 2))) + raw[12:]
```

Comment:
It works but this implementation is a little baffling to read. I'd think it would read better if we checked that `raw` is 12 characters long to begin with. Also, (although this is from a different commit) the `reduce` statement would be more readable as `':'.join(...)`. I tend to agree with Guido Van Rossum on how to use `reduce` in Python, though we may disagree on other things.